frontend/src/views/FarmInformation/FarmInformation.tsx

```diff
@@ -58,6 +55,41 @@ export default function FarmInformation() {
     { value: 3, label: 'Columbia Shuswap' },
   ];
 
+  const handleSubmit = () => {
+    const data = state.nmpFile;
+    let nmpFile: NMPFile;
+    if (data) {
+      try {
+        nmpFile = JSON.parse(data);
+
+        if (!nmpFile.farmDetails) {
+          nmpFile.farmDetails = {};
+        }
+        nmpFile.farmDetails.Year = formData.Year;
```

Comment:
Why does it matter whether the data is parsed or not? In both cases, everything gets overriden by the formData.
Also you can definitely combine the last part of the if-else into one part. What you would do is:
`nmpFile.farmDetails = { ...nmpFile.farmDetails, [insert all of the setting here]}`

Reply:
@raarielgrace Fixed the logic here :)